File: CurvedBottomNavigationBar/MainActivity.java
package com.tonevellah.meowbottomnavigationbar;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;

import android.os.Bundle;
import android.widget.Toast;

import com.etebarian.meowbottomnavigation.MeowBottomNavigation;

public class MainActivity extends AppCompatActivity {
    //member
    MeowBottomNavigation bottomNav;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        //assign the variable
        bottomNav = findViewById(R.id.bottomNav);

        //add menu items to bottom nav
        bottomNav.add(new MeowBottomNavigation.Model(1, R.drawable.ic_home));
        bottomNav.add(new MeowBottomNavigation.Model(2, R.drawable.ic_search));
        bottomNav.add(new MeowBottomNavigation.Model(3, R.drawable.ic_person));
        bottomNav.add(new MeowBottomNavigation.Model(4, R.drawable.ic_settings));

        //set bottom nav on show listener
        bottomNav.setOnShowListener(new MeowBottomNavigation.ShowListener() {
            @Override
            public void onShowItem(MeowBottomNavigation.Model item) {
                //define a fragment
                Fragment fragment;

                //initialize fragment according to its id
                 if (item.getId() ==4){
                     fragment = new SettingsFragment();
                 }else  if (item.getId() ==3){
                     fragment = new UserFragment();
                 }
                 else  if (item.getId() ==2){
                     fragment = new SearchFragment();
                 }
                 else {
                     fragment = new HomeFragment();
                 }

                 //use load fragment method to show the current fragment
                loadFragment(fragment);
            }
        });

        //set the initial fragment to show
        bottomNav.show(1, true);

        //set menu item on click listener
        bottomNav.setOnClickMenuListener(new MeowBottomNavigation.ClickListener() {
            @Override
            public void onClickItem(MeowBottomNavigation.Model item) {
                //display a toast
                Toast.makeText(getApplicationContext()," You clicked "+ item.getId(), Toast.LENGTH_SHORT).show();
            }
        });

     //set on reselect listener
        bottomNav.setOnReselectListener(new MeowBottomNavigation.ReselectListener() {
            @Override
            public void onReselectItem(MeowBottomNavigation.Model item) {
                //display a toast
                Toast.makeText(getApplicationContext()," You reselected "+ item.getId(), Toast.LENGTH_SHORT).show();
            }
        });

        //set count to dashboard item
        bottomNav.setCount(3, "10");

    }


    //define a load method to feed the screen
    private void loadFragment(Fragment fragment) {
        //replace the fragment
        getSupportFragmentManager()
                .beginTransaction()
                .replace(R.id.nav_host_fragment_container,fragment, null)
                .commit();
    }
}

File: SplashScreen/MainActivity.kt
package com.tonevellah.splashscreenapi

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.Handler
import android.view.View
import android.view.ViewTreeObserver
import androidx.core.splashscreen.SplashScreen.Companion.installSplashScreen
import com.tonevellah.splashscreenapi.R

class MainActivity : AppCompatActivity() {
    //define a boolean variable
    var isAndroidReady = false
    override fun onCreate(savedInstanceState: Bundle?) {
        //handle the splash screen transitions
        installSplashScreen()

        //keep splash screen for a long time
        val content = findViewById<View>(android.R.id.content)
        //add the pre draw listener to the view tree observer
        content.viewTreeObserver.addOnPreDrawListener(object : ViewTreeObserver.OnPreDrawListener {
            override fun onPreDraw(): Boolean {
                if (isAndroidReady) {
                    content.viewTreeObserver.removeOnPreDrawListener(this)
                }
                //define a method to alter isAndroidReady boolean
                dismissSplashScreen()
                return false
            }
        })
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    private fun dismissSplashScreen() {
        //use an handler
        Handler().postDelayed({ //change the boolean
            isAndroidReady = true
        }, 5000)
    }
}
